lib/logstash/filters/jdbc/lookup.rb

```diff
@@ -82,10 +82,11 @@ def formatted_errors
 
     def enhance(local, event)
       if @prepared_statement
-        result = call_prepared(local, event)
+        load_method_ref = method(:load_data_from_prepared)
       else
-        result = fetch(local, event) # should return a LookupResult
+        load_method_ref = method(:load_data_from_local)
       end
```

Comment:
could we move this whole block to happen during `LookupProcessor#initialize`?
`enhance` is called on `plugin.filter` so we end up doing the method lookup every time.
we should be able to keep the looked up method in an instance field as `@prepared_statement` is known ahead.